### PR TITLE
MOVE-64: secure sr

### PR DESCRIPTION
**Pull request by martinwam**

### CODE REVIEW

File: serviceregistry-server/pom.xml [hunk 1]
```diff
@@ -199,6 +199,16 @@
         </dependency>
 
         <dependency>
+            <groupId>org.springframework.security.oauth</groupId>
+            <artifactId>spring-security-oauth2</artifactId>
+        </dependency>
+
+        <dependency>
```
Comment (nuth) [inline serviceregistry-server/pom.xml]: Trenger vi denne i service registry?

File: serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordController.java [hunk 2]
```diff
@@ -28,6 +29,7 @@
 @RequestMapping("/identifier")
 @ExposesResourceFor(EntityResource.class)
 @RestController
+@PreAuthorize("#oauth2.hasAnyScope('scope_move_1', 'scope_move_2', 'move/dpo.read', 'move/dpv.read', 'move/dpi.read')")
```
Comment (nuth) [inline serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordController.java]: Her burde vi flytte følgende:
```
        if (usesSikkerDigitalPost().test(entityInfo)) {
            entity.setServiceRecord(serviceRecordFactory.createSikkerDigitalPostRecord(identifier));
        }
        if (usesFormidlingstjenesten().test(entityInfo)) {
            entity.setServiceRecord(serviceRecordFactory.createEduServiceRecord(identifier));
        }
        if (usesPostTilVirksomhet().test(entityInfo)) {
            entity.setServiceRecord(serviceRecordFactory.createPostVirksomhetServiceRecord(identifier));
        }
```
ut i en slags service med en metode per type, annotert med `@PreAuthorize` på metode nivå og riktig scope.

File: serviceregistry-server/src/main/resources/application-staging.properties [hunk 3]
```diff
@@ -5,4 +5,8 @@ difi.move.dpi.endpointURL=https://qaoffentlig.meldingsformidler.digipost.no/api/
 difi.move.krr.client.alias=client_alias
 difi.move.krr.client.location=cloud:kontaktinfo-client-test.jks
 difi.move.krr.server.alias=ver2
-difi.move.krr.server.location=cloud:kontaktinfo-server-test.jks
\ No newline at end of file
+difi.move.krr.server.location=cloud:kontaktinfo-server-test.jks
+difi.move.auth.enable=false
+security.oauth2.resource.token-info-uri=https://eid-exttest.difi.no/idporten-oidc-provider/tokeninfo
```
Comment (nuth) [inline serviceregistry-server/src/main/resources/application-staging.properties]: Er dette riktig for staging?

Comment (martinwam) [inline serviceregistry-server/src/main/resources/application-staging.properties]: Inntil vi får en ev. prod-versjon, ja.

**Comment by nuth** [comment]
SonarQube analysis reported 3 issues
* CRITICAL 2 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: jackson-core-asl-1.9.13.jar | Reference: CVE-2016-3720 | CVSS Score: 7.5 | XML external entity (XXE) vulnerability in XmlMapper in the Data format extension for Jackson (aka jackson-dataformat-xml) allows attackers to have unspecified impact via unknown vectors. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: jackson-mapper-asl-1.9.13.jar | Reference: CVE-2016-3720 | CVSS Score: 7.5 | XML external entity (XXE) vulnerability in XmlMapper in the Data format extension for Jackson (aka jackson-dataformat-xml) allows attackers to have unspecified impact via unknown vectors. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)

**Pull request merged**